server/src/main/java/io/crate/expression/symbol/SymbolVisitors.java

```diff
@@ -217,6 +217,18 @@ protected Void visitSymbol(Symbol symbol, Void context) {
             }
             return null;
         }
+
+        @Override
+        public Void visitSelectSymbol(SelectSymbol selectSymbol, Void context) {
+            if (haystack.contains(selectSymbol)) {
+                consumer.accept((T) selectSymbol);
+            } else {
+                for (Symbol output : selectSymbol.relation().outputs()) {
+                    output.accept(this, context);
+                }
```

Comment:
This is the new logic. The check if `selectSymbol` was part of the haystack was supported before, but if not we will check for symbols within the `selectSymbol`. This makes sure `SELECT b FROM (foo))` can match `foo.b`.

Reply:
I'm not sure if it doesn't get a bit too confusing if the intersection method starts crossing relation boundaries
The CorrelatedJoin operator already contains some logic to do that: 

https://github.com/crate/crate/blob/d8c0c5a6e89677765a996f40ab9bee9ddafc9b7f/server/src/main/java/io/crate/planner/operators/CorrelatedJoin.java#L154-L160

Couldn't that be extended or modified instead?